### ISSUE TITLE
Handle failing SourceForge mirrors for gcc

### ISSUE THREAD

**Issue by twogood** [lukesampson/scoop#2161]
My Appveyor build fails right now because of:
```
Installing 'gcc' (7.2.0) [64bit]
The remote server returned an error: (502) Bad Gateway.
URL https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/7.2.0/threads-posix/seh/x86_64-7.2.0-release-posix-seh-rt_v5-rev1.7z is not valid
```

See https://github.com/twogood/unshield/pull/76

**Comment by r15ch13** [comment]
The download itself works. SourceForge is quite the problem child...
Currently, the only solution would be to simply retry it. 😞

**Comment by twogood** [comment]
Actually, I can't get that URL to work on my desktop machine either, but the same gcc version worked on March 10.

A bit off topic but maybe Scoop itself could support mirrors or fallback URLs in the future?

**Comment by asmgf** [comment]
It may be related to #2109 since we don't use `downloads.sourceforge.net` for GCC.

I don't see any error though, Scoop just loads a web-page instead of the archive and fails on `Checking hash`.

Changing the `-imatch` to a more generic one or changing the URL to `https://downloads.sourceforge.net/project/mingw-w64/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/7.2.0/threads-posix/seh/x86_64-7.2.0-release-posix-seh-rt_v5-rev1.7z` fixes the issue.

**Comment by twogood** [comment]
Using downloads.sourceforge.net instead sounds like a good idea!

**Comment by rasa** [comment]
Fixed in #2168

**Comment by twogood** [comment]
Works fine now. You are awesome!
